src/oge/output_data.py

```diff
@@ -669,47 +672,31 @@ def write_power_sector_results(
                 # concat the totals to the fuel-specific totals
                 ba_table = pd.concat([ba_table, ba_total], axis=0, ignore_index=True)
 
-            # aggregate data to monthly
-            ba_table_monthly = (
-                ba_table.groupby(["fuel_category", "report_date"], dropna=False)
-                .sum(numeric_only=True)
-                .reset_index()
-            )
-            ba_table_monthly = add_generated_emission_rate_columns(ba_table_monthly)
-            # re-order columns
-            ba_table_monthly = ba_table_monthly[
-                ["fuel_category", "report_date"]
-                + DATA_COLUMNS
-                + GENERATED_EMISSION_RATE_COLS
-            ]
-            output_to_results(
-                ba_table_monthly,
-                year,
-                ba,
-                "power_sector_data/monthly/",
-                path_prefix,
-                skip_outputs,
-            )
+                agg = {}
+                if include_monthly:
+                    agg["monthly"] = ["fuel_category", "report_date"]
 
-            # aggregate data to annual
-            ba_table_annual = (
-                ba_table.groupby(["fuel_category"], dropna=False)
-                .sum(numeric_only=True)
-                .reset_index()
-            )
-            ba_table_annual = add_generated_emission_rate_columns(ba_table_annual)
-            # re-order columns
-            ba_table_annual = ba_table_annual[
-                ["fuel_category"] + DATA_COLUMNS + GENERATED_EMISSION_RATE_COLS
-            ]
-            output_to_results(
-                ba_table_annual,
-                year,
-                ba,
-                "power_sector_data/annual/",
-                path_prefix,
-                skip_outputs,
-            )
+                if include_annual:
+                    agg["annual"] = ["fuel_category"]
+
+                for k, v in agg.items():
```

Comment:
instead of k,v can we make these descriptive variables like agg_level, groupby_cols